examples/echo_server.hs

```diff
@@ -0,0 +1,18 @@
+{-# OPTIONS -XOverloadedStrings #-}
+
+import qualified Network.Freddy as Freddy
+import Data.ByteString.Lazy.Char8 (ByteString)
+
+processMessage :: Freddy.Request -> IO ()
+processMessage (Freddy.Request body replyWith failWith) = replyWith body
+
+main = do
+  fConnection <- Freddy.connect "127.0.0.1" "/" "guest" "guest"
```

Comment:
Also generally other monads.
You usually want to `map`/`flatMap` over them and avoid passing them around in functions too much,
because passing it around makes the whole chain very hard to understand.

It's the same reason why it's a bad idea to pass around observables between objects (except for simple binding).